maxim/tests/mock_writer.py

```
@@ -0,0 +1,260 @@
+"""
+Mock writer for testing purposes.
+
+This module provides a mock implementation of LogWriter that captures
+logs in memory for testing purposes.
+"""
+
+import threading
+from queue import Queue
+from typing import List
```

Ruff (0.11.9)

`typing.List` is deprecated, use `list` instead

(UP035)